common/lib/dependabot/clients/codecommit.rb

```diff
@@ -11,26 +11,17 @@ class NotFound < StandardError; end
       # Constructor methods #
       #######################
 
-      def self.for_source(source:, credentials:)
-        credential =
-          credentials.
-          select { |cred| cred["type"] == "git_source" }.
-          find { |cred| cred["region"] == source.hostname }
-
-        new(source, credential)
+      def self.for_source(source:)
+        new(source)
       end
 
       ##########
       # Client #
       ##########
 
-      def initialize(source, credentials)
+      def initialize(source)
         @source = source
-        @cc_client = Aws::CodeCommit::Client.new(
-          access_key_id: credentials&.fetch("username"),
-          secret_access_key: credentials&.fetch("password"),
-          region: credentials&.fetch("region")
-        )
```

Comment:
As written you _can't_ specify a session token unless it's explicitly passed to the codecommit client on instantiation.  However, if we don't pass anything to the codecommit client (directly) we can fall back to the aws sdk and let _it_ worry about credentials: 
> When :credentials are not configured directly, the following locations will be searched for credentials:
> 
> Aws.config[:credentials]
The :access_key_id, :secret_access_key, and :session_token options.
ENV['AWS_ACCESS_KEY_ID'], ENV['AWS_SECRET_ACCESS_KEY']
~/.aws/credentials
~/.aws/config
EC2 IMDS instance profile - When used by default, the timeouts are very aggressive. Construct and pass an instance of Aws::InstanceProfileCredentails to enable retries and extended timeouts.

[Source](https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/CodeCommit/Client.html#initialize-instance_method)

To answer your question more directly though: I'm passing in env vars to the dependabot docker image to get these values set. Based on the pattern used in `dependabot-script` and `bin\dry-run.rb` to build credentials I don't _think_ this will break anything

Reply:
@userhas404d ah yeah those scripts would work if you passed env vars but we don't require this so would break for users who've set the credentials in code only. Could you not pull the session token from the credentials like is done for `region` and then populate it from the env var?

`session_token: credentials&.fetch("session_token")`

Reply:
@feelepxyz not directly related to codecommit but this PR addresses some similar concerns regarding aws credentials and how they're handled with the aws sdk: https://github.com/dependabot/dependabot-core/pull/1463#issuecomment-545930970

While you _can_ recreate all the input vars the aws sdk expects externally, it's significantly easier to just fall back to the local env/aws configuration. I'd argue that if anyone was using this integration they're already an aws heavy shop and would expect the "aws credentials provided from the env" type of functionality in tools that run against aws resources.

Reply:
@userhas404d sorry for the super slow review on this! Still a bit wary of breaking this interface for existing integrations. What do you think of passing credentials to this method like before but fall back to `@cc_client = Aws::CodeCommit::Client.new` if credentials is `nil`?